File: bonsai_scxml_engine/src/main/java/de/unibi/citec/clf/bonsai/engine/scxml/config/MissingSlotFault.java
package de.unibi.citec.clf.bonsai.engine.scxml.config;

import de.unibi.citec.clf.bonsai.engine.model.StateID;

/**
 * @author lruegeme
 */
public class MissingSlotFault {

    private static final long serialVersionUID = 5250728310328591284L;
    private StateID state;
    private String slotKey;
    private boolean error = true;

    MissingSlotFault(String slotKey, StateID state) {
        this.slotKey = slotKey;
        this.state = state;
    }

    MissingSlotFault(String slotKey, StateID state, boolean error) {
        this.slotKey = slotKey;
        this.state = state;
        this.error = error;
    }

    public boolean isError() {
        return error;
    }

    public String getMessage() {
        if (error) {
            return "Missing slot definition with key \"" + slotKey + "\" for state \"" + state.getCanonicalID() + "\"";
        } else {
            return "Default slot definition with key \"" + slotKey + "\" for state \"" + state.getCanonicalID() + "\"" + " xpath is " + "\\" + slotKey;
        }
    }

    public StateID getState() {
        return state;
    }

    public String getSlotKey() {
        return slotKey;
    }

}

File: bonsai_scxml_engine/src/main/java/de/unibi/citec/clf/bonsai/engine/config/fault/MissingSlotFault.kt
package de.unibi.citec.clf.bonsai.engine.config.fault

import de.unibi.citec.clf.bonsai.engine.model.StateID
import java.util.*

/**
 * @author lruegeme
 */
class MissingSlotFault {
    var state: StateID
        private set
    var slotKey: String
        private set
    var type: Class<*> = Objects::class.java
        private set
    var isError = true
        private set

    constructor(slotKey: String, state: StateID, type: Class<*>) {
        this.slotKey = slotKey
        this.state = state
        this.type = type
    }

    constructor(slotKey: String, state: StateID, type: Class<*>, error: Boolean) {
        this.slotKey = slotKey
        this.state = state
        this.type = type
        isError = error
    }

    val message: String
        get() = if (isError) {
            "Missing slot definition with key '$slotKey' for state '${state.getCanonicalID()}' of type '$type'"
        } else {
            "Default slot definition with key '$slotKey' for state '${state.getCanonicalID()}' of type '$type' xpath is $slotKey"
        }

    companion object {
        private const val serialVersionUID = 5250728310328591284L
    }
}
